bee-ledger/src/workers/consensus/white_flag.rs

```diff
@@ -0,0 +1,288 @@
+    }
+
+    metadata.balance_diffs.merge(balance_diffs)?;
+
+    Ok(ConflictReason::None)
+}
+
+async fn apply_transaction<B: StorageBackend>(
+    storage: &B,
+    message_id: &MessageId,
+    transaction: &TransactionPayload,
+    metadata: &mut WhiteFlagMetadata,
+) -> Result<ConflictReason, Error> {
+    match transaction.essence() {
+        Essence::Regular(essence) => {
+            apply_regular_essence(
+                storage,
+                message_id,
+                &transaction.id(),
+                essence,
+                transaction.unlock_blocks(),
+                metadata,
+            )
+            .await
+        }
+        essence => Err(Error::UnsupportedTransactionEssenceKind(essence.kind())),
+    }
+}
+
+async fn apply_message<B: StorageBackend>(
+    storage: &B,
+    message_id: &MessageId,
+    message: &Message,
+    metadata: &mut WhiteFlagMetadata,
+) -> Result<(), Error> {
+    metadata.referenced_messages += 1;
+
+    match message.payload() {
+        Some(Payload::Transaction(transaction)) => {
+            match apply_transaction(storage, message_id, transaction, metadata).await? {
+                ConflictReason::None => metadata.included_messages.push(*message_id),
+                conflict => metadata.excluded_conflicting_messages.push((*message_id, conflict)),
+            }
+        }
+        _ => metadata.excluded_no_transaction_messages.push(*message_id),
+    }
+
+    Ok(())
+}
+
+async fn traversal<B: StorageBackend>(
+    tangle: &MsTangle<B>,
+    storage: &B,
+    mut message_ids: Vec<MessageId>,
+    metadata: &mut WhiteFlagMetadata,
+) -> Result<(), Error> {
+    let mut visited = HashSet::new();
+
+    while let Some(message_id) = message_ids.last() {
+        match tangle
+            .get_vertex(&message_id)
+            .await
+            .as_ref()
+            .and_then(|v| v.message_and_metadata().cloned())
+        {
+            Some((message, meta)) => {
+                if meta.flags().is_referenced() {
+                    visited.insert(*message_id);
+                    message_ids.pop();
+                    continue;
+                }
+
+                if let Some(unvisited) = message.parents().iter().find(|p| !visited.contains(p)) {
+                    message_ids.push(*unvisited);
+                } else {
+                    apply_message(storage, message_id, &message, metadata).await?;
+                    visited.insert(*message_id);
+                    message_ids.pop();
+                }
+            }
+            None => {
+                if !tangle.is_solid_entry_point(message_id).await {
+                    return Err(Error::MissingMessage(*message_id));
+                } else {
+                    visited.insert(*message_id);
+                    message_ids.pop();
+                }
+            }
+        }
+    }
+
+    Ok(())
+}
+
+/// Computes the ledger state according to the White Flag method.
+pub async fn white_flag<B: StorageBackend>(
```

Comment:
Maybe not now, but I think in the future we should be more descriptive here, what the `White Flag` method is actually. Would be nice to read it broken down into 1 or 2 single sentences what is being done in this function according to White Flag. For now a link to the White Flag RFC would be a nice addition to the doc comment.